spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/RabbitOperationsInvokingAdvice.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.support;
+
+import java.time.Duration;
+
+import org.aopalliance.intercept.MethodInvocation;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.amqp.rabbit.core.RabbitOperations;
+import org.springframework.integration.handler.advice.HandleMessageAdvice;
+import org.springframework.lang.Nullable;
+
+/**
+ * An advice that executes all downstream {@link RabbitOperations} operations
+ * on the same channel, as long as there are no thread handoffs.
+ * The same RabbitOperations must be used in this and all downstream components.
+ * Typically used with a splitter or some other mechanism that would cause
+ * multiple messages to be sent.
+ *
+ * @author Gary Russell
+ * @since 5.1
+ *
+ */
+public class RabbitOperationsInvokingAdvice implements HandleMessageAdvice {
```

Comment:
M-m-m. Maybe just `RabbitOperationsAdvice`? Because that our `OperationsCallback` really assumes to let us to call all the possible `RabbitOperations` during `invoke()`

Reply:
Hmmm `RabbitOperationsAdvice` implies it advises the operations.

Reply:
Just `RabbitChannelAdvice`? With the meaning that we wrap the method with a dedicated thread-bound Rabbit channel. 
Well, anything works for me since I understand what is this advice about. We just need to come up with some fluent name to let end users to determine the purpose clearly or at least close.

Sorry for being so nit-pick, but for me, as non-English speaker, all those `ing`-ings are hard to translate since there is no such a structure in our language :smile: 

I think we will stop with the next your name. It took already a plenty of time to discuss almost for nothing.

Reply:
No; I think it's worth spending time to get it right.

`RabbitBoundChannelAdvice`
`RabbitBindChannelAdvice`
`RabbitReserveChannelAdvice`
`BindRabbitChannelAdvice`
`BoundRabbitChannelAdvice`

I think I like the last one - describes what it does - creates a channel that is bound to the thread.

IMHO we have too many "bindings" - SCSt, RabbitMQ bindings, so I think we should avoid Bind, Binding in the name.